.github/workflows/docker.yaml

```diff
@@ -0,0 +1,55 @@
+name: Build Docker
+
+on:
+  push:
+    tags: ["v*"]
+  pull_request:
+    paths:
+      - Dockerfile
+      - docker-bake.hcl
+      - ".github/workflows/docker.yaml"
+  schedule:
+    - cron: "44 4 * * *"
+  workflow_dispatch:
+
+jobs:
+  docker:
+    name: Build Docker
+    runs-on: ubuntu-24.04
+    strategy:
+      matrix:
+        postgres: ["15", "16", "17"]
+        platform: ["linux/amd64", "linux/arm64"]
+
+    steps:
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: pgduckdb
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Checkout pg_duckdb extension code
+        uses: actions/checkout@v4
+        with:
+          submodules: "recursive"
+      - name: Set env
+        run: |
+          # find the first tag for this commit starting with 'v'
+          echo "POSTGRES_VERSION=${{ matrix.postgres }}" >> $GITHUB_ENV
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker buildx
+        uses: docker/setup-buildx-action@v3
+        with:
+          platforms: ${{ matrix.platform }}
+      - name: docker bake
+        uses: docker/bake-action@v5
+        with:
+          targets: pg_duckdb_${{ matrix.postgres }}
+          # push: ${{ github.event_name != 'pull_request' }}
+          push: true
+          set: |
+            *.platform=${{ matrix.platform }}
+            *.cache-to=type=gha,mode=max
+            *.cache-from=type=gha
+            postgres.tags=pgduckdb/pgduckdb:${{ matrix.postgres }}-${{ github.sha }}
+            ${{ !contains(github.ref_name, '/') && format('postgres.tags=pgduckdb/pgduckdb:{0}-{1}', matrix.postgres, github.ref_name) || '' }}
```

Comment:
Will this override the one above or add it? Also instead of skipping when branch name have a `/` could we replace with `-`? (seems arbitrary to just exclude these no?)

Reply:
It adds to the one above it, so we will push with the sha anytime it gets built, including PRs.

Mm yeah we could replace `/` with `-`, but this was sorta an accidental "feature" -- by doing it this way we only get tags and `main` pushed as "named tags" rather than PRs.

Reply:
That won't work for tags though (`v0.1.0`)